File: frontend/src/api/category.ts
import request from '../utils/request'

export interface CategoryItem {
  id: string
  name: string
  icon: string | null
  sort_order: number
  status: string
  device_count: number
}

export interface CategoryCreate {
  name: string
  icon?: string
  sort_order?: number
}

export interface CategoryUpdate {
  name?: string
  icon?: string
  sort_order?: number
}

export function getCategories() {
  return request.get('/categories')
}

export function createCategory(data: CategoryCreate) {
  return request.post('/categories', data)
}

export function updateCategory(id: string, data: CategoryUpdate) {
  return request.put(`/categories/${id}`, data)
}

export function deleteCategory(id: string) {
  return request.delete(`/categories/${id}`)
}

File: backend/app/schemas/category.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    icon: str | None = None
    sort_order: int = 0


class CategoryUpdate(BaseModel):
    name: str | None = None
    icon: str | None = None
    sort_order: int | None = None


class CategoryResponse(BaseModel):
    id: UUID
    name: str
    icon: str | None = None
    sort_order: int
    status: str
    device_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}